### PR TITLE
AYS-523 | Architecture Has Been Refactored and Sonar Issuses Have Been Fixed

### PR DESCRIPTION
**Pull request by agitrubard**
### Checklist

Before submitting your pull request, ensure the following:

- [X] **Title and Branch Naming Conventions:**
    - The pull request title follows the
      standard: [Pull Request Naming Conventions](https://github.com/afet-yonetim-sistemi/ays-be/blob/main/CONTRIBUTING.md#pull-request-naming-conventions).
    - The branch name follows one of the conventions outlined in
      the [Branch Naming Conventions](https://github.com/afet-yonetim-sistemi/ays-be/blob/main/CONTRIBUTING.md#branch-naming-conventions).

- [X] **Local Testing:**
    - I have tested my changes locally on Postman, and they are working as expected.

- [X] **Code Quality:**
    - The code is formatted according to the project's coding guidelines and style.
    - The code has been reviewed to ensure its quality.
    - The code does not contain any issues flagged by SonarLint.

- [X] **Documentation:**
    - Necessary documentation has been added or existing documentation has been updated, specifically detailing changes made in Postman.

- [X] **Testing:**
    - Relevant unit tests have been written and included.
    - Relevant integration tests have been written and included.

- [X] **Reviewers and Assignees:**
    - Default reviewers have been assigned to this pull request.
    - Assignees have been added if necessary.

- [X] **Labels and Associations:**
    - No specific actions are required in the Labels and Associations section for this pull request.

### CODE REVIEW

File: src/main/java/org/ays/common/exception/AysConflictException.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
+package org.ays.common.exception;
+
+import java.io.Serial;
+
+/**
+ * Represents an exception that occurs when a conflict is detected in the AYS (Are You Sure) system.
```
Comment (MenekseYuncu) [inline src/main/java/org/ays/common/exception/AysConflictException.java:6]: AYS (Are You Sure) system  :D

File: src/main/java/org/ays/common/model/AysPageable.java [hunk 2]
```diff
@@ -31,7 +31,7 @@ public class AysPageable extends AysSort {
      * Represents the page number.
      * This value should be between 1 and 99999999.
      */
-    @Range(min = 1, max = 99999999)
+    @Range(min = 1, max = 99_999_999)
```
Comment (MenekseYuncu) [inline src/main/java/org/ays/common/model/AysPageable.java:34]: @Range(min = 1, max = 99_999_999) 
max neden bu şekilde tanımlandı?

File: src/main/java/org/ays/common/exception/AysConflictException.java (hunk 1 above)
Comment (agitrubard) [reply src/main/java/org/ays/common/exception/AysConflictException.java:6]: AI Generation Issue 👀

File: src/main/java/org/ays/common/model/AysPageable.java (hunk 2 above)
Comment (agitrubard) [reply src/main/java/org/ays/common/model/AysPageable.java:34]: Bu konuyla ilgili issue paylaşacaktım buradan önden paylaşmış olayım 😄 
image

File: src/main/java/org/ays/common/exception/AysConflictException.java (hunk 1 above)
Comment (agitrubard) [reply src/main/java/org/ays/common/exception/AysConflictException.java]: @MenekseYuncu Fixed ✅

File: src/main/java/org/ays/common/model/AysPageable.java (hunk 2 above)
Comment (MenekseYuncu) [reply src/main/java/org/ays/common/model/AysPageable.java:34]: hee iyiymiş :D

**Pull request merged**